File: src/backendOld/portsIn/store/IStore.cs
namespace SnippetVault {
using System;
using System.Threading.Tasks;


public interface IStore {
    Task<ReqResult<SnippetDTO>> snippetsGet(int taskGroupId, int lang1Id, int lang2Id);
    Task<ReqResult<SnippetDTO>> snippetsGetByCode(string taskGroupCode, string lang1Code, string lang2Code);
    Task<ReqResult<SnippetIntern>> snippetGet(int snId);
    Task<ReqResult<LanguageDTO>> languagesGet();
    Task<ReqResult<TaskGroupDTO>> taskGroupsGet();
    Task<ReqResult<ProposalDTO>> proposalsGet();
    Task<ReqResult<TaskDTO>> tasksFromGroupGet(int taskGroup);
    Task<ReqResult<TaskGroupDTO>> taskGroupsForLangGet(int langId);
    Task<ReqResult<TaskGroupDTO>> taskGroupsForLangsGet(int lang1, int lang2);
    Task<ReqResult<AlternativeDTO>> alternativesForTLGet(int taskLanguageId);
    Task<ReqResult<AlternativeDTO>> alternativesForUserGet(int taskLanguageId, int userId);
    Task<ReqResult<CommentDTO>> commentsGet(int snippetId);
    Task<int> proposalCreate(ProposalCreateDTO dTO, int authorId);
    Task<int> proposalUpdate(ProposalUpdateDTO dto);

    Task<int> snippetApprove(int sn);
    Task<int> snippetDecline(int sn);
    Task<int> snippetMarkPrimary(int tlId, int snId);
    Task<ReqResult<TaskDTO>> taskGet(int taskId);
    Task<ReqResult<TaskDTO>> taskForTLGet(int taskLanguageId);

    Task<ReqResult<TaskCUIntern>> tasksAll();
    Task<int> taskGroupCU(TaskGroupCUDTO dto);
    Task<ReqResult<TaskGroupCUDTO>> taskGroupsAll();
    Task<ReqResult<LanguageCUDTO>> languagesAll();
    Task<int> taskCU(TaskCUDTO dto);
    Task<int> languageCU(LanguageCUDTO dto);
    Task<ReqResult<StatsDTO>> statsForAdmin();
    Task<long> userCount();

    Task<ReqResult<AuthenticateIntern>> userAuthentGet(string userName);
    Task<ReqResult<AuthorizeIntern>> userAuthorizGet(int userId);
    Task<ReqResult<AuthorizeIntern>> userAdminAuthoriz();
    Task<int> userUpdateExpiration(int userId, string newToken, DateTime newDate);
    Task<int> userRegister(UserNewIntern user);
    Task<int> userUpdate(UserNewIntern user);
    Task<int> userVote(int userId, int tlId, int snId);
    Task<ReqResult<ProfileDTO>> userProfile(int userId);
    Task<ReqResult<UserDTO>> userData(int userId);
    Task<int> commentCreate(int userId, int snId, string content, DateTime ts);

    Task<int> logMessage(DateTime ts, int msgType, string code, string message);
}

}

File: src/backend/main/tech/sozonov/SnippetVault/portsIn/IStore.java
package tech.sozonov.SnippetVault.portsIn;
import java.time.LocalDateTime;
import reactor.core.publisher.Mono;
import tech.sozonov.SnippetVault.core.DTO.AuxDTO.*;
import tech.sozonov.SnippetVault.core.DTO.SnippetDTO.*;
import tech.sozonov.SnippetVault.core.DTO.UserDTO.*;
import tech.sozonov.SnippetVault.core.internal.InternalTypes.*;
import tech.sozonov.SnippetVault.core.utils.Types.ReqResult;


public interface IStore {

Mono<ReqResult<Snippet>> snippetsGet(int taskGroupId, int lang1Id, int lang2Id);
Mono<ReqResult<Snippet>> snippetsGetByCode(String taskGroupCode, String lang1Code, String lang2Code);
Mono<ReqResult<SnippetIntern>> snippetGet(int snId);
Mono<ReqResult<Language>> languagesGet();
Mono<ReqResult<TaskGroup>> taskGroupsGet();
Mono<ReqResult<Proposal>> proposalsGet();
Mono<ReqResult<Task>> tasksFromGroupGet(int taskGroup);
Mono<ReqResult<TaskGroup>> taskGroupsForLangGet(int langId);
Mono<ReqResult<TaskGroup>> taskGroupsForLangsGet(int lang1, int lang2);
Mono<ReqResult<Alternative>> alternativesForTLGet(int taskLanguageId);
Mono<ReqResult<Alternative>> alternativesForUserGet(int taskLanguageId, int userId);
Mono<ReqResult<Comment>> commentsGet(int snippetId);
Mono<Integer> proposalCreate(ProposalCreate dto, int authorId);
Mono<Integer> proposalUpdate(ProposalUpdate dto);

Mono<Integer> snippetApprove(int sn);
Mono<Integer> snippetDecline(int sn);
Mono<Integer> snippetMarkPrimary(int tlId, int snId);
Mono<ReqResult<Task>> taskGet(int taskId);
Mono<ReqResult<Task>> taskForTLGet(int taskLanguageId);

Mono<ReqResult<TaskCUIntern>> tasksAll();
Mono<Integer> taskGroupCU(TaskGroupCU dto);
Mono<ReqResult<TaskGroupCU>> taskGroupsAll();
Mono<ReqResult<LanguageCU>> languagesAll();
Mono<Integer> taskCU(TaskCU dto);
Mono<Integer> languageCU(LanguageCU dto);
Mono<ReqResult<Stats>> statsForAdmin();
Mono<Long> userCount();

Mono<ReqResult<AuthenticateIntern>> userAuthentGet(String userName);
Mono<ReqResult<AuthorizeIntern>> userAuthorizGet(int userId);
Mono<ReqResult<AuthorizeIntern>> userAdminAuthoriz();
Mono<Integer> userUpdateExpiration(int userId, String newToken, LocalDateTime newDate);
Mono<Integer> userRegister(UserNewIntern user);
Mono<Integer> userUpdate(UserNewIntern user);
Mono<Integer> userVote(int userId, int tlId, int snId);
Mono<ReqResult<Profile>> userProfile(int userId);
Mono<ReqResult<User>> userData(int userId);
Mono<Integer> commentCreate(int userId, int snId, String content, LocalDateTime ts);

Mono<Integer> logMessage(LocalDateTime ts, int msgType, String code, String message);

}
